guides/authentication.md

```diff
@@ -0,0 +1,210 @@
+Authentication in browsers
+==========================
+
+To utilize authentication in browsers using the SDK you need to use the `createClientWithOAuth` function from the SDK.
+
+`createClientWithOAuth` will automatically handle authentication of the user and respond when the client get a `401`-response (not logged in) response from the API.
+
+# Table of Contents
+- [Authentication in browsers](#authentication-in-browsers)
+- [Table of Contents](#table-of-contents)
+  - [Why not using api keys?](#why-not-using-api-keys)
+  - [Access tokens](#access-tokens)
+  - [Authentication flow](#authentication-flow)
+  - [How to authenticate with the SDK?](#how-to-authenticate-with-the-sdk)
+    - [Authentication with redirects](#authentication-with-redirects)
+      - [Simple example](#simple-example)
+      - [Customize redirect URL](#customize-redirect-url)
+    - [Authentication with popups](#authentication-with-popups)
+      - [Simple example](#simple-example-1)
+      - [Customize redirect URL](#customize-redirect-url-1)
+  - [Advance](#advance)
+    - [Manually trigger authentication](#manually-trigger-authentication)
+    - [Skip authentication](#skip-authentication)
+    - [Combine different authentication methods](#combine-different-authentication-methods)
+    - [Tokens](#tokens)
+  - [More](#more)
+
+## Why not using api keys?
+
+It is **strongly recommended** to not use api keys in web applications since the api key is easily readable by everyone that have access to the application. Another restriction is that all users of your app share the same api-key. This means that all users have the same access level.
+
+## Access tokens
+
+The solution to avoid using api keys is to use access tokens. This is short living tokens which grants the user access to CDF. The application get an access token by asking the user to login to the IdP-provier (Google, Active Directory etc) of a CDF project.
```

Comment:
```suggestion
The solution to avoid using api keys is to use access tokens. This is short living tokens which grants the user access to CDF. The application get an access token by asking the user to login to the identity provider (Google, Active Directory etc) of a CDF project.
```